### PR TITLE
NIFI-2395: Ensure that if we fail to index provenance events we do not …

### PR DESCRIPTION
**Pull request by markap14**
…prevent the repo from continuing to merge journals

### CODE REVIEW

File: nifi-nar-bundles/nifi-provenance-repository-bundle/nifi-persistent-provenance-repository/src/main/java/org/apache/nifi/provenance/PersistentProvenanceRepository.java [hunk 1]
```diff
@@ -1668,24 +1669,32 @@ public Thread newThread(final Runnable r) {
...
                     try {
                         for (int i = 0; i < configuration.getIndexThreadPoolSize(); i++) {
                             final Callable<Object> callable = new Callable<Object>() {
                                 @Override
                                 public Object call() throws IOException {
                                     while (!eventQueue.isEmpty() || !finishedAdding.get()) {
-                                        final Tuple<StandardProvenanceEventRecord, Integer> tuple;
                                         try {
-                                            tuple = eventQueue.poll(10, TimeUnit.MILLISECONDS);
-                                        } catch (final InterruptedException ie) {
```
Comment (joewitt) [inline nifi-nar-bundles/nifi-provenance-repository-bundle/nifi-persistent-provenance-repository/src/main/java/org/apache/nifi/provenance/PersistentProvenanceRepository.java]: shouldn't we still catch interrupted and clear the intererupt?

Comment (markap14) [inline nifi-nar-bundles/nifi-provenance-repository-bundle/nifi-persistent-provenance-repository/src/main/java/org/apache/nifi/provenance/PersistentProvenanceRepository.java]: yeah you're right - we need to ensure that we re-set the interrupt flag. Pushed an update to the PR.

File: nifi-nar-bundles/nifi-provenance-repository-bundle/nifi-persistent-provenance-repository/src/main/java/org/apache/nifi/provenance/PersistentProvenanceRepository.java [hunk 2]
```diff
@@ -1668,24 +1669,32 @@ public Thread newThread(final Runnable r) {
...
-                                            continue;
+                                            final Tuple<StandardProvenanceEventRecord, Integer> tuple;
+                                            try {
+                                                tuple = eventQueue.poll(10, TimeUnit.MILLISECONDS);
+                                            } catch (final InterruptedException ie) {
+                                                continue;
```
Comment (joewitt) [inline nifi-nar-bundles/nifi-provenance-repository-bundle/nifi-persistent-provenance-repository/src/main/java/org/apache/nifi/provenance/PersistentProvenanceRepository.java]: @markap14  we want a "Thread.currentThread().interrupt();" here as well yes?

Comment (markap14) [inline nifi-nar-bundles/nifi-provenance-repository-bundle/nifi-persistent-provenance-repository/src/main/java/org/apache/nifi/provenance/PersistentProvenanceRepository.java]: D'oh! You caught another one! Updated.

**Pull request closed without merging**